File: HackerRank/Week of Code 37/Simple Language.cpp
#include <bits/stdc++.h>
using namespace std;
long long int maximumProgramValue(long int n)
{
  string t;
  long long int y, x = 0, temp = 0;
  while (n--)
  {
    cin >> t >> y;
    if (t == "add")
    {
      x = x + y;
      if (x >= temp)
        temp = x;
    }
    else if (t == "set")
    {
      x = y;
      if (x >= temp)
        temp = x;
    }
  }
  return temp;
}

int main()
{
  long int n;
  cin >> n;
  long long int result = maximumProgramValue(n);
  cout << result << "\n";
  return 0;
}

File: HackerRank/Week of Code 37/Simple Language.py
#!/bin/python3

import os
import sys


#
# Complete the maximumProgramValue function below.
#
def maximumProgramValue(n):
    #
    # Write your code here.
    #
    x = 0
    temp = 0

    for i in range(n):
        t = str(input())
        y = int(input())
        if t is "add":
            x += y
            if temp < x:
                temp = x
        elif t is "set":
            x = y
            if temp < x:
                temp = x

    return temp


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    result = maximumProgramValue(n)

    fptr.write(str(result) + '\n')

    fptr.close()
